src/Database/PostgreSQL/PQTypes/SQL/Builder.hs

```diff
@@ -518,32 +533,42 @@ sqlDelete table refine =
                                , sqlDeleteWhere  = []
                                , sqlDeleteResult = []
                                , sqlDeleteWith   = []
+                               , sqlDeleteRecursiveWith = NonRecursive
                                })
 
 
 data Materialized = Materialized | NonMaterialized
+data Recursive = Recursive | NonRecursive
 
-class SqlWith a where
-  sqlWith1 :: a -> SQL -> SQL -> Materialized -> a
+instance Semigroup Recursive where
```

Comment:
You can put a comment here that explains why it works like this :slightly_smiling_face: